velox/vector/tests/VectorTest.cpp

```diff
@@ -2089,3 +2262,178 @@ TEST_F(VectorTest, createVectorWithNullType) {
           pool(), nullptr, nullptr, 100, nullptr, nullptr, nullptr, nullptr),
       kErrorMessage);
 }
+
+TEST_F(VectorTest, dataDependentFlags) {
```

Comment:
We have dedicated tests for ensureWritable and prepareForReuse APIs: velox/vector/tests/EnsureWritableVectorTest.cpp and velox/vector/tests/VectorPrepareForReuseTest.cpp. Consider moving new test cases there.